contracts/libraries/SortedDoublyLL.sol

```diff
@@ -0,0 +1,359 @@
+        }
+
+        self.nodes[_id].key = _key;
+
+        if (prevId == address(0) && nextId == address(0)) {
+            // Insert as head and tail
+            self.head = _id;
+            self.tail = _id;
+        } else if (prevId == address(0)) {
+            // Insert before `prevId` as the head
+            self.nodes[_id].nextId = self.head;
+            self.nodes[self.head].prevId = _id;
+            self.head = _id;
+        } else if (nextId == address(0)) {
+            // Insert after `nextId` as the tail
+            self.nodes[_id].prevId = self.tail;
+            self.nodes[self.tail].nextId = _id;
+            self.tail = _id;
+        } else {
+            // Insert at insert position between `prevId` and `nextId`
+            self.nodes[_id].nextId = nextId;
+            self.nodes[_id].prevId = prevId;
+            self.nodes[prevId].nextId = _id;
+            self.nodes[nextId].prevId = _id;
+        }
+
+        self.size++;
+    }
+
+    /*
+     * @dev Remove a node from the list
+     * @param _id Node's id
+     */
+    function remove(Data storage self, address _id) public {
+        // List must contain the node
+        require(contains(self, _id));
+
+        if (self.size > 1) {
+            // List contains more than a single node
+            if (_id == self.head) {
+                // The removed node is the head
+                // Set head to next node
+                self.head = self.nodes[_id].nextId;
+                // Set prev pointer of new head to null
+                self.nodes[self.head].prevId = address(0);
+            } else if (_id == self.tail) {
+                // The removed node is the tail
+                // Set tail to previous node
+                self.tail = self.nodes[_id].prevId;
+                // Set next pointer of new tail to null
+                self.nodes[self.tail].nextId = address(0);
+            } else {
+                // The removed node is neither the head nor the tail
+                // Set next pointer of previous node to the next node
+                self.nodes[self.nodes[_id].prevId].nextId = self.nodes[_id].nextId;
+                // Set prev pointer of next node to the previous node
+                self.nodes[self.nodes[_id].nextId].prevId = self.nodes[_id].prevId;
+            }
+        } else {
+            // List contains a single node
+            // Set the head and tail to null
+            self.head = address(0);
+            self.tail = address(0);
+        }
+
+        delete self.nodes[_id];
+        self.size--;
+    }
+
+    /*
+     * @dev Increase the key of a node in the list
+     * @param _id Node's id
+     * @param _amount Amount to increase node's key
+     * @param _prevId Id of previous node for the new insert position
+     * @param _nextId Id of next node for the new insert position
+     */
+    function increaseKey(Data storage self, address _id, uint256 _amount, address _prevId, address _nextId) public {
```

Comment:
I was just thinking that maybe the user of the DoublyLL shouldn't have to know whether they are increasing or decreasing the key, they should just call `updateKey()`? What do you think?